### PR TITLE
Rename static route local_pref to rib_priority

### PR DESCRIPTION
**Pull request by taspelund**
This is the omicron half of maghemite#359. This renames the `local_pref`
field of static routes to `rib_priority` to avoid overloading the name
of a standardized and well-known BGP attribute (Local Preference). This
change shrinks the size of the field from u32 -> u8 and unwraps the
Option<T> before shipping routes to mgd, aligning with the updated API.
This also brings omicron back into sync with maghemite.

Fixes: #6711

Signed-off-by: Trey Aspelund <trey@oxidecomputer.com>

### CODE REVIEW

**Comment by ahl** [comment]
Can you please address #6711 in this PR as well?

**Comment by rcgoodfellow** [comment]
Something potentially helpful here is this script

- https://github.com/oxidecomputer/omicron/blob/main/tools/update_maghemite.sh

**Review by internet-diglett — APPROVED** [review]

**Comment by rcgoodfellow** [comment]
Noting that because this is changing the external API, there will need to be a corresponding update to oxide.rs

**Review by rcgoodfellow — APPROVED** [review]

File: nexus/src/app/background/tasks/sync_switch_configuration.rs [hunk 1]
```diff
@@ -71,6 +71,10 @@ const PHY0: &str = "phy0";
 // before breaking to check for shutdown conditions.
 const BGP_SESSION_RESOLUTION: u64 = 100;
 
+// This is the default RIB Priority used for static routes.  This mirrors
+// the const defined in maghemite in rdb/src/lib.rs.
+const DEFAULT_RIB_PRIORITY_STATIC: u8 = 1;
```
Comment (zeeshanlakhani) [inline nexus/src/app/background/tasks/sync_switch_configuration.rs:76]: Should this be imported from the rdb lib (where it's pub)?

**Review by zeeshanlakhani — COMMENTED** [review]
This mostly looks straightforward and good, but I had just one comment/Q.

File: nexus/src/app/background/tasks/sync_switch_configuration.rs (hunk 1 above)
Comment (taspelund) [reply nexus/src/app/background/tasks/sync_switch_configuration.rs:76]: Yeah, definitely.  I'll address this in a follow-up PR, so we can get this merged + omicron/maghemite back in sync for R11.

File: nexus/types/src/external_api/params.rs [hunk 2]
```diff
@@ -1695,7 +1695,7 @@ pub struct Route {
 
     /// Local preference for route. Higher preference indictes precedence
     /// within and across protocols.
```
Comment (ahl) [inline nexus/types/src/external_api/params.rs:1697]: @taspelund should this have been updated?

File: sled-agent/types/src/early_networking.rs [hunk 3]
```diff
@@ -323,8 +323,8 @@ pub mod back_compat {
         pub uplink_cidr: Ipv4Net,
         /// VLAN id to use for uplink
         pub uplink_vid: Option<u16>,
-        /// Local preference
-        pub local_pref: Option<u32>,
+        /// RIB Priority
+        pub rib_priority: Option<u8>,
```
Comment (ahl) [inline sled-agent/types/src/early_networking.rs:327]: I see that this struct was deprecated; was updating it the right thing to do?

Comment (rcgoodfellow) [reply sled-agent/types/src/early_networking.rs:327]: Good catch. @internet-diglett can you take a look at this PR from the perspective of bootstore compatibility?

Comment (internet-diglett) [reply sled-agent/types/src/early_networking.rs:327]: In theory it should be harmless at this point, but ahl is correct that it probably shouldn't have been updated.

**Pull request merged**